src/api/java/com/minecolonies/api/util/ItemStackUtils.java

```diff
@@ -303,51 +303,34 @@ else if (Compatibility.isTinkersTool(stack, toolType))
             return Compatibility.getToolLevel(stack);
         }
 
-        if (ToolType.HOE.equals(toolType))
-        {
-            if (stack.getItem() instanceof HoeItem)
-            {
-                final HoeItem hoeItem = (HoeItem) stack.getItem();
-                return hoeItem.getTier().getLevel();
-            }
-        }
-        else if (ToolType.SWORD.equals(toolType))
-        {
-            if (stack.getItem() instanceof SwordItem)
-            {
-                final SwordItem SwordItem = (SwordItem) stack.getItem();
-                return SwordItem.getTier().getLevel();
-            }
-
-        }
-        else if (ToolType.HELMET.equals(toolType)
-                   || ToolType.BOOTS.equals(toolType)
-                   || ToolType.CHESTPLATE.equals(toolType)
-                   || ToolType.LEGGINGS.equals(toolType))
+        if (ToolType.HELMET.equals(toolType)
+                || ToolType.BOOTS.equals(toolType)
+                || ToolType.CHESTPLATE.equals(toolType)
+                || ToolType.LEGGINGS.equals(toolType))
         {
             if (stack.getItem() instanceof ArmorItem)
             {
                 final ArmorItem ArmorItem = (ArmorItem) stack.getItem();
                 return getArmorLevel(ArmorItem.getMaterial());
             }
         }
-        else if (stack.getItem() instanceof FishingRodItem)
+        else if (stack.getItem() instanceof TieredItem)  // most tools
+        {
+            return ((TieredItem) stack.getItem()).getTier().getLevel();
+        }
+        else if (toolType.equals(ToolType.FISHINGROD))
         {
             return getFishingRodLevel(stack);
         }
         else if (toolType.equals(ToolType.SHEARS))
         {
-            return stack.getItem() instanceof ShearsItem ? 0 : -1;
+            return 0;
```

Comment:
I remember that without this piece, results were sometimes iffy

Reply:
It's already done the `isTool` check by this point, so it can't reach here unless that decided it was valid shears.  (That in turn requires an item that does both `SHEARS_DIG` and `SHEARS_HARVEST`, but does not require that it be a `ShearsItem`.  Perhaps notably we also don't check for `SHEARS_CARVE` or `SHEARS_DISARM` either, but the latter is unlikely to be appropriate and while the former kind of is, it seems a little unnecessary to check everything.)

The rearrangement of the code does mean that it might sometimes return a value higher than 0, if for example a mod introduces new `TieredItem` shears with different material levels.  But if `isTool` passed then there's no reason for this to return -1.